test/client.test.js

```diff
@@ -196,3 +196,244 @@ test('should emit error when invalid feature runtime', t => {
+                {
+                    name: 'unleash-internal-group-strategy',
+                    group,
+                },
+            ]);
+        },
+    };
+    const strategies = [new CustomFalseStrategy(), new CustomStrategy()];
+    const client = new Client(repo, strategies);
+    client.on('error', log).on('warn', log);
+    return client.isEnabled('feature');
+}
+
+test('should return the group value for sub lists when all is true', t => {
+    const result = testGroup([{ name: 'custom', operator: 'AND' }, { name: 'custom' }]);
+
+    t.true(result === true);
+
+    const result2 = testGroup([
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom' },
+    ]);
+
+    t.true(result2 === true);
+
+    const result3 = testGroup([{ name: 'custom' }]);
+
+    t.true(result3 === true);
+});
+
+test('group value is false and has 2 entries and OR default operator', t => {
+    const result = testGroup([{ name: 'custom-false' }, { name: 'custom-false' }]);
+
+    t.true(result === false);
+
+    const result2 = testGroup([
+        { name: 'custom' },
+        { name: 'custom-false' },
+        { name: 'custom-false' },
+    ]);
+
+    t.true(result2 === true);
+
+    const result3 = testGroup([
+        { name: 'custom-false' },
+        { name: 'custom' },
+        { name: 'custom-false' },
+    ]);
+
+    t.true(result3 === true);
+});
+
+test('group value is false and has 2 entries', t => {
+    const result = testGroup([{ name: 'custom', operator: 'AND' }, { name: 'custom-false' }]);
+
+    t.true(result === false);
+
+    const result2 = testGroup([{ name: 'custom-false', operator: 'AND' }, { name: 'custom' }]);
+
+    t.true(result2 === false);
+});
+
+test('groups value with one false value and multiple entries', t => {
+    const result = testGroup([
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom-false' },
+    ]);
+
+    t.true(result === false);
+
+    const result2 = testGroup([
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom-false', operator: 'AND' },
+        { name: 'custom' },
+    ]);
+
+    t.true(result2 === false);
+
+    const result3 = testGroup([
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom-false', operator: 'AND' },
+        { name: 'custom', operator: 'AND' },
+        { name: 'custom' },
+    ]);
+
+    t.true(result3 === false);
+});
+
+test('groups with inner AND group that resovles to true', t => {
+    const result = testGroup([
+        {
+            name: 'inner-group',
+            group: [{ name: 'custom', operator: 'AND' }, { name: 'custom' }],
+            operator: 'AND',
```

Comment:
Does it ever make sense to have `operator` defined on the group-level if the code only looks on it on strategy-definition level inside groups?

Reply:
I think we initially should target to not use operator on the top-level, but further down the line I dont see any point on supporting legacy.

Reply:
agreed :+1: